File: problems/longest_subarray_of_1's_after_deleting_one_element/solution.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int start=0,mx=0, count=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==0){
                count++;
            }
            while(count>1){
                if(nums[start]==0){
                    count--;
                }
                start++;
            }
            mx = max(mx,i-start);
        }
        return mx;
    }
};

File: problems/longest_subarray_of_1's_after_deleting_one_element/solution.java
class Solution {
    public static int longestSubarray(int[] nums) {
        boolean zero = false;
        int start = 0;
        int end = 0;
        int max = Integer.MIN_VALUE;
        for(;end<nums.length;end++){
            if(nums[end]==0 && zero){

                while(nums[start]!=0){
                    start+=1;
                }
                start++;
            }
            else if(nums[end]==0){
                zero=true;
            }
            // System.out.println("start: "+start);
            // System.out.println("end: "+end);
            max = Math.max(max, end-start);
        }
        return max;
    }
}